### PR TITLE
Case PATCH_METADATA

### PR DESCRIPTION
**Pull request by pawbu**

### CODE REVIEW

File: datastore_version_manager/adapter/built_datasets.py [hunk 1]
```diff
@@ -14,6 +15,21 @@ def get_metadata_path(dataset_name: str) -> str:
         )
 
 
+def get_metadata(dataset_name: str) -> dict:
```
Comment (DanielElisenberg) [inline datastore_version_manager/adapter/built_datasets.py:18]: :100:

File: datastore_version_manager/api/request_models.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ class NewPendingOperationRequest(BaseModel, extra=Extra.forbid):
 
 class UpdatePendingOperationRequest(BaseModel, extra=Extra.forbid):
     releaseStatus: str
-    operationType: Optional[str]
+    description: Optional[str]
```
Comment (DanielElisenberg) [inline datastore_version_manager/api/request_models.py:14]: :100:

File: datastore_version_manager/domain/draft_dataset.py [hunk 3]
```diff
@@ -6,26 +6,29 @@
...
         if datastore.is_dataset_in_datastore_versions(dataset_name, "RELEASED"):
             operation_type = 'CHANGE_DATA'
         else:
             operation_type = 'ADD'
 
+    if operation_type == 'PATCH_METADATA':
+        if not datastore.is_dataset_in_datastore_versions(dataset_name, "RELEASED"):
+            raise ForbiddenOperation(
+                f'Cannot patch variable "{dataset_name}". '
+                'This variable has not been released.'
```
Comment (DanielElisenberg) [inline datastore_version_manager/domain/draft_dataset.py:29]: Should be consistently name them variables or datasets?

File: datastore_version_manager/domain/draft_dataset.py [hunk 4]
```diff
@@ -38,31 +41,55 @@ def add_new_draft_dataset(operation_type: str,
...
+    built_metadata = built_datasets.get_metadata(dataset_name)
+    patched = metadata_patcher.patch(built_metadata, latest_metadata)
+    built_datasets.write_metadata(patched, built_metadata_path)
+
+
+def check_if_dataset_was_deleted(dataset_name):
+    if datastore.is_dataset_in_datastore_versions(dataset_name, "DELETED"):
+        raise ForbiddenOperation(
+            f'Cannot add or update variable "{dataset_name}". '
+            'This variable has been removed from datastore'
```
Comment (DanielElisenberg) [inline datastore_version_manager/domain/draft_dataset.py:81]: :100: same logging comment [variable | dataset]

File: datastore_version_manager/domain/pending_operations.py [hunk 5]
```diff
@@ -1,25 +1,17 @@
...
 
-    dataset_in_pending_operations = any(
-        dataset['name'] == dataset_name for dataset in data_structure_updates
-    )
-    if dataset_in_pending_operations:
-        raise PendingOperationException(
-            f'Cannot add new pending operation for {dataset_name} because it already exists. '
-            'Use update or hard delete.'
-        )
+    check_if_dataset_not_already_present(data_structure_updates, dataset_name)
```
Comment (DanielElisenberg) [inline datastore_version_manager/domain/pending_operations.py:14]: I would like the function name to be more explicit. ```check_if_dataset_in_pending_operations``` or similar
alternatively remove the 'not'

File: tests/unit/domain/test_metadata_patcher.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+def teardown_function():
+    shutil.rmtree('tests/resources')
+    shutil.move(
+        'tests/resources_backup',
+        'tests/resources'
+    )
+
+
+def test_patch_metadata():
+    # source_json has changes in fields that will be skipped (temporality)
```
Comment (DanielElisenberg) [inline tests/unit/domain/test_metadata_patcher.py:25]: I assume this means we only support updating populationDescription, but in the future will support updating things like codeList.categoryTitle?

in v1.5 do we only support changing the populationDescription(or what that field would be called in v1.5)?

**Review by DanielElisenberg — APPROVED** [review]
Looks good :100: Like that you updated the updatePendingOperation logic at the same time 🚀 

Some minor readability comments and questions.
Are we thinking a new PR for adding the codeList stuff soon, or should we keep this in the backlog for now (as it represents a "new feature")?

File: datastore_version_manager/domain/draft_dataset.py (hunk 3 above)
Comment (pawbu) [reply datastore_version_manager/domain/draft_dataset.py:29]: I checked the code and it looks like the messages to the user use the term "variable" whereas source code uses the term "dataset" consistently across the project. I thinks it's OK this way.

**Comment by pawbu** [comment]
> Looks good 💯 Like that you updated the updatePendingOperation logic at the same time 🚀
> 
> Some minor readability comments and questions. Are we thinking a new PR for adding the codeList stuff soon, or should we keep this in the backlog for now (as it represents a "new feature")?

I'm going to add a seprate task for this as we're awaiting user input.

File: datastore_version_manager/domain/draft_dataset.py (hunk 4 above)
Comment (pawbu) [reply datastore_version_manager/domain/draft_dataset.py:81]: Same response :)

File: tests/unit/domain/test_metadata_patcher.py (hunk 6 above)
Comment (pawbu) [reply tests/unit/domain/test_metadata_patcher.py:25]: Yes, supporting more fields will be a part of a separate task.

In 1.5 we support changing more fields, but in 2.0 there were changes to the model that resulted in deprecation of most of them.

**Pull request merged**